src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -581,6 +581,41 @@ public async Task GoAwayFrame_NewRequest_NewConnection()
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task GoAwayFrame_ServerDisconnect_AbortStreams()
+        {
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (HttpClient client = CreateHttpClient())
+            {
+                server.AllowMultipleConnections = true;
+
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address);
+                Http2LoopbackConnection connection = await server.EstablishConnectionAsync();
+                int streamId = await connection.ReadRequestHeaderAsync();
+
+                await connection.SendGoAway(streamId);
+
+                // wait until GOAWAY received
+                await connection.PingPong();
+                await connection.WaitForConnectionShutdownAsync();
+
+                // Timeout is small on purpose, connection should be closed immediatelly
+                // We add some slack though in case machine is being slow
+                const int ShortTimeout = 2000;
```

Comment:
Is this 2 seconds?  That's unfortunately been shown to not actually be very long on a loaded machine in CI.  We generally use at least 30s for things we expect to happen but want to allow some wiggleroom for.

Reply:
@stephentoub 30s will be hard to tell the difference between hanging and cancelled by test framework vs cancelled because server got disposed. In theory this should happen instantly and 2s felt like enough although I'm not sure if enough for CI.

In theory the exception type differs but I'm not sure if that is always a case

Reply:
I can almost guarantee you that if you leave it at 2s, you'll soon get an issue opened about the test failing.  At that point, what action will you take?